editor/src/components/canvas/floating-panels.tsx

```diff
@@ -0,0 +1,754 @@
+          ) {
+            // dragging to far left when something is already there
+            shouldSwitchPanel = 'leftMenu2'
+          }
+          if (
+            droppedToOutsideOfAColumn.targetPanel === 'leftMenu2' &&
+            leftColumn2.length > 0 &&
+            (leftColumn.length === 0 || (draggedPanel === 'leftMenu1' && leftColumn.length === 1))
+          ) {
+            // dragging to the right of the left panel 2
+            shouldSwitchPanel = 'leftMenu1'
+          }
+
+          if (
+            droppedToOutsideOfAColumn.targetPanel === 'rightMenu1' &&
+            rightColumn.length > 0 &&
+            (rightColumn2.length === 0 ||
+              (draggedPanel === 'rightMenu2' && rightColumn2.length === 1))
+          ) {
+            // dragging to the left of the right panel 1
+            shouldSwitchPanel = 'rightMenu2'
+          }
+
+          if (
+            droppedToOutsideOfAColumn.targetPanel === 'rightMenu2' &&
+            rightColumn2.length > 0 &&
+            (rightColumn.length === 0 ||
+              (draggedPanel === 'rightMenu1' && rightColumn.length === 1))
+          ) {
+            // dragging to far right when something is already there
+            shouldSwitchPanel = 'rightMenu1'
+          }
+
+          return {
+            newPanel: droppedToOutsideOfAColumn.targetPanel,
+            switchWithPanel: shouldSwitchPanel,
+          }
+        }
+
+        return {
+          newPanel:
+            panelsData.find((panel) => {
+              if (panel.frame == null) {
+                return null
+              }
+              return rectContainsPoint(panel.frame, newPosition)
+            })?.location ?? null,
+          switchWithPanel: null,
+        }
+      })()
+
+      const currentPanelData = panelsData.find((v) => v.name === draggedMenuOrPane)
+      if (
+        currentPanelData != null &&
+        newPanel != null &&
+        (draggedPanel != newPanel || switchWithPanel != null)
+      ) {
+        const columnData = panelsData.filter((d) => d.location === newPanel)
+        let updatedPanelData = panelsData.map((panel) => {
+          if (panel.name === draggedMenuOrPane) {
+            return {
+              ...panel,
+              location: newPanel,
+              locationInColumn: columnData.length,
+            }
+          } else if (panel.location === newPanel && switchWithPanel != null) {
+            return { ...panel, location: switchWithPanel }
+          } else if (draggedPanel === panel.location) {
+            return {
+              ...panel,
+              locationInColumn:
+                panel.locationInColumn > currentPanelData.locationInColumn
+                  ? panel.locationInColumn - 1
+                  : panel.locationInColumn,
+            }
+          } else {
+            return panel
+          }
+        })
+
+        // Readjust the locationInColumn values to have them start at zero and increment by 1.
+        updatedPanelData.sort(
```

Comment:
This is a patch fix to correct this for now without making any dramatic last minute changes.